pkg/services/contexthandler/contexthandler.go

```diff
@@ -111,7 +111,9 @@ func (h *ContextHandler) Middleware(next http.Handler) http.Handler {
 			reqContext.Logger = reqContext.Logger.New("traceID", traceID)
 		}
 
-		identity, err := h.authnService.Authenticate(reqContext.Req.Context(), &authn.Request{HTTPRequest: reqContext.Req, Resp: reqContext.Resp})
+		ctx = trace.ContextWithSpan(reqContext.Req.Context(), span)
+
+		identity, err := h.authnService.Authenticate(ctx, &authn.Request{HTTPRequest: reqContext.Req, Resp: reqContext.Resp})
```

Comment:
@gamab noticed you was the last one refactoring this so would be good to get some additional eyes in case I've done anything bad :)

Reply:
if im reading this correctly.. maybe we could do 

```
		authnCtx, authnSpan := h.tracer.Start(reqContext.Req.Context(), "Auth - Middleware")
		identity, err := h.authnService.Authenticate(authnCtx, &authn.Request{HTTPRequest: reqContext.Req, Resp: reqContext.Resp})
```

+ remove line 88 to emphasize that this context should be used just for authn?

Reply:
We could, but then it would happen a bit later than in the beginning of the middleware I guess. I'm open ears for suggestions

Reply:
It makes sense that this span would not wrap anything under it, we already have a span that wraps the whole request.

I think we should either do what @ArturWierzbicki suggested so we propagate this span to Authenticat or we can also just remove this span. [Authenticate](https://github.com/grafana/grafana/blob/0585b55bb1c1fe83b721e4607fe3a9e8d354776e/pkg/services/authn/authnimpl/service.go#L89-L124) is already creating a span that wraps all authentication logic for a request so this one seems a bit unnecessary

Reply:
Thanks. I would suggest keeping it as is since nice to include the middleware as well and it might change in the future to do more things. I was more curious from your perspective that we don't change the context somehow that's passed to `Authenticate` and screw things up? :)